src/range-slider/index.tsx

```diff
@@ -63,281 +62,253 @@ export interface RangeSliderProperties extends ThemedProperties {
+					themeCss.thumb,
+					themeCss.leftThumb,
+					focus.isFocused('slider1') ? themeCss.focused : undefined,
+					fixedCss.thumbFixed
+				]}
+				styles={{
+					left: Math.round(slider1Percent * 100) + '%'
+				}}
+			/>
+			<div
+				key="rightThumb"
+				classes={[
+					themeCss.thumb,
+					themeCss.rightThumb,
+					focus.isFocused('slider2') ? themeCss.focused : undefined,
+					fixedCss.thumbFixed
+				]}
+				styles={{
+					left: Math.round(slider2Percent * 100) + '%'
+				}}
+			/>
+			{showOutput ? (
+				<output
+					classes={[themeCss.output, outputIsTooltip ? themeCss.outputTooltip : null]}
+					for={widgetId}
+					styles={
+						outputIsTooltip
+							? {
+									left: `${Math.round(
+										(slider1Percent + (slider2Percent - slider1Percent) / 2) *
+											100
+									)}%`
+							  }
+							: undefined
 					}
-				},
-				[
-					slider1,
-					v(
-						'div',
-						{
-							key: 'minimumLabel',
-							classes: [baseCss.visuallyHidden],
-							id: this._minLabelId
-						},
-						[minimumLabel]
-					),
-					slider2,
-					v(
-						'div',
-						{
-							key: 'maximumLabel',
-							classes: [baseCss.visuallyHidden],
-							id: this._maxLabelId
-						},
-						[maximumLabel]
-					),
-					v('div', {
-						key: 'track',
-						classes: [this.theme(css.filled), fixedCss.filledFixed],
-						styles: {
-							left: Math.round(slider1Percent * 100) + '%',
-							width: Math.round((slider2Percent - slider1Percent) * 100) + '%'
-						}
-					}),
-					v('div', {
-						key: 'leftThumb',
-						classes: [
-							...this.theme([
-								css.thumb,
-								css.leftThumb,
-								slider1Focus.active ? css.focused : undefined
-							]),
-							fixedCss.thumbFixed
-						],
-						styles: {
-							left: Math.round(slider1Percent * 100) + '%'
-						}
-					}),
-					v('div', {
-						key: 'rightThumb',
-						classes: [
-							...this.theme([
-								css.thumb,
-								css.rightThumb,
-								slider2Focus.active ? css.focused : undefined
-							]),
-							fixedCss.thumbFixed
-						],
-						styles: {
-							left: Math.round(slider2Percent * 100) + '%'
-						}
-					}),
-					showOutput
-						? this.renderOutput({ min, max }, [slider1Percent, slider2Percent])
-						: null
-				]
-			)
-		];
+					tabIndex={-1}
+				>
+					{output ? output({ min, max }) : `${min}, ${max}`}
+				</output>
+			) : null}
+		</div>
+	];
 
-		return v(
-			'div',
-			{
-				key: 'root',
-				classes: this.theme(this.getRootClasses())
-			},
-			labelAfter ? children.reverse() : children
-		);
-	}
-}
+	return (
+		<div
+			key="root"
+			classes={[
+				themeCss.root,
+				disabled ? themeCss.disabled : null,
+				focus.isFocused('root') ? themeCss.focused : null,
+				valid === false ? themeCss.invalid : null,
+				valid === true ? themeCss.valid : null,
+				readOnly ? themeCss.readonly : null,
+				showOutput ? themeCss.hasOutput : null
+			]}
+		>
+			{labelAfter ? children.reverse() : children}
```

Comment:
we've dropped `labelAfter`, I don't think we want it here anymore either, label should go above.